pkg/config/process.go

```diff
@@ -0,0 +1,74 @@
+func procBindEnvAndSetDefault(config Config, key string, val interface{}, env ...string) {
+	// Uppercase, replace "." with "_" and add "DD_" prefix to key so that we follow the same environment
+	// variable convention as the core agent.
+	processConfigKey := "DD_" + strings.Replace(strings.ToUpper(key), ".", "_", -1)
+	processAgentKey := strings.Replace(processConfigKey, "PROCESS_CONFIG", "PROCESS_AGENT", 1)
+
+	envs := append([]string{processConfigKey, processAgentKey}, env...)
+	config.BindEnvAndSetDefault(key, val, envs...)
+}
+
+func setupProcesses(config Config) {
+	config.SetDefault("process_config.enabled", "false")
+	// process_config.enabled is only used on Windows by the core agent to start the process agent service.
+	// it can be set from file, but not from env. Override it with value from DD_PROCESS_AGENT_ENABLED.
+	ddProcessAgentEnabled, found := os.LookupEnv("DD_PROCESS_AGENT_ENABLED")
+	if found {
+		AddOverride("process_config.enabled", ddProcessAgentEnabled)
+	}
+
+	config.BindEnv("process_config.process_dd_url", "")
+
+	config.SetKnown("process_config.dd_agent_env")
+	config.SetKnown("process_config.enabled")
+	config.SetKnown("process_config.intervals.process_realtime")
+	config.SetKnown("process_config.queue_size")
+	config.SetKnown("process_config.rt_queue_size")
+	config.SetKnown("process_config.max_per_message")
+	config.SetKnown("process_config.max_ctr_procs_per_message")
+	config.SetKnown("process_config.cmd_port")
+	config.SetKnown("process_config.intervals.process")
+	config.SetKnown("process_config.blacklist_patterns")
+	config.SetKnown("process_config.intervals.container")
+	config.SetKnown("process_config.intervals.container_realtime")
+	procBindEnvAndSetDefault(config, "process_config.dd_agent_bin", DefaultDDAgentBin)
+	config.SetKnown("process_config.custom_sensitive_words")
+	config.SetKnown("process_config.scrub_args")
+	config.SetKnown("process_config.strip_proc_arguments")
+	config.SetKnown("process_config.windows.args_refresh_interval")
+	config.SetKnown("process_config.windows.add_new_args")
+	config.SetKnown("process_config.windows.use_perf_counters")
+	config.SetKnown("process_config.additional_endpoints.*")
+	config.SetKnown("process_config.container_source")
+	config.SetKnown("process_config.intervals.connections")
+	config.SetKnown("process_config.expvar_port")
+	procBindEnvAndSetDefault(config, "process_config.log_file", DefaultProcessAgentLogFile)
+	config.SetKnown("process_config.internal_profiling.enabled")
+	procBindEnvAndSetDefault(config, "process_config.grpc_connection_timeout_secs", DefaultGRPCConnectionTimeoutSecs)
+	procBindEnvAndSetDefault(config, "process_config.remote_tagger", true)
+
+	// Process Discovery Check
+	procBindEnvAndSetDefault(config, "process_config.process_discovery.enabled", false, "DD_PROCESS_CONFIG_DISCOVERY_ENABLED", "DD_PROCESS_AGENT_DISCOVERY_ENABLED")
```

Comment:
It would be best to replace it with `config.BindEnvAndSetDefault` directly to avoid confusion, see comment earlier.

Reply:
Fixed!